src/tasks/EmitBundleTask/EmitBundleBase.cs

```diff
@@ -0,0 +1,381 @@
+    /// the bundled resources.
+    /// </summary>
+    public string? BundleFile { get; set; }
+
+    /// <summary>
+    /// The filename for the generated header file that declares
+    /// MonoBundled*Resource struct types.
+    /// </summary>
+    public string BundleHeader { get; set; } = "mono-bundled-source.h";
+
+    /// <summary>
+    /// Filename for the unified source containing all byte data and len
+    /// of the FilesToBundle resources.
+    /// Leave empty to keep sources separate.
+    /// </summary>
+    public string? CombinedResourceSource { get; set; }
+
+    /// <summary>
+    /// Path to store build artifacts
+    /// <summary>
+    public string OutputDirectory {get; set; } = default!;
+
+    /// <summary>
+    /// Resources that were bundled
+    ///
+    /// Successful bundling will set the following metadata on the items:
+    ///   - DataSymbol
+    ///   - LenSymbol
+    /// <summary>
+    [Output]
+    public ITaskItem[] BundledResources { get; set; } = default!;
+
+    public override bool Execute()
+    {
+        // The DestinationFile (output filename) already includes a content hash. Grouping by this filename therefore
+        // produces one group per file-content. We only want to emit one copy of each file-content, and one symbol for it.
+        var filesToBundleByDestinationFileName = FilesToBundle.GroupBy(f => f.GetMetadata("DestinationFile")).ToList();
+
+        // We're handling the incrementalism within this task, because it needs to be based on file content hashes
+        // and not on timetamps. The output filenames contain a content hash, so if any such file already exists on
+        // disk with that name, we know it must be up-to-date.
+        var remainingDestinationFilesToBundle = filesToBundleByDestinationFileName.Where(g => !File.Exists(g.Key)).ToArray();
+
+        // If you're only touching the leaf project, we don't really need to tell you that.
+        // But if there's more work to do it's valuable to show progress.
+        var verbose = remainingDestinationFilesToBundle.Length > 1;
+        var verboseCount = 0;
+
+        var filesToBundleByRegisteredName = FilesToBundle.GroupBy(file => {
+            var registeredName = file.GetMetadata("RegisteredName");
+            if(string.IsNullOrEmpty(registeredName))
+            {
+                registeredName = Path.GetFileName(file.ItemSpec);
+            }
+            return registeredName;
+        }).ToList();
+
+        var files = filesToBundleByRegisteredName.Select(group => {
+            var registeredFile = group.First();
+            var outputFile = registeredFile.GetMetadata("DestinationFile");
+            var registeredFilename = group.Key;
+            var resourceName = ToSafeSymbolName(outputFile);
+            string culture = registeredFile.GetMetadata("Culture");
+            string? resourceSymbolName = null;
+            if (File.Exists(registeredFile.GetMetadata("Symfile")))
```

Comment:
Other metadata uses full names and File, so maybe this metadata should be SymbolFile?

Reply:
Changed all symfiles to symbol/symbol data.